removes the remainder from ScalerFor method

p0lyn0mial opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
4. If the PR is unfinished, see how to mark it: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
this PR removes existing scalers from `ScalerFor` method

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
`kubectl scale` can now scale any resource (kube, CRD, aggregate) conforming to the standard scale endpoint
```

p0lyn0mial commented:
/assign @deads2k @DirectXMan12

p0lyn0mial commented:
An intermediary step that removes existing scalers from `ScalerFor` method. This is just to narrow the scope in case of potential issues.

deads2k commented on pkg/kubectl/scale.go:
```diff
@@ -167,6 +153,7 @@ func (precondition *ScalePrecondition) ValidateReplicationController(controller
 	return nil
 }
 
+// TODO(p0lyn0mial): remove ReplicationControllerScaler
```
Being held for the reapers?  If so, open an issue.

p0lyn0mial replied:
Yes, that is correct. I was planning to deal with the reapers in the very next pull. In that case do we need an issue for that ?

Unless your plan is to leave the reapers for the time being and focus on preparing kubectl to support scaling custom resources.

deads2k commented on pkg/kubectl/scale.go:
```diff
@@ -54,21 +53,8 @@ type Scaler interface {
-	default:
-		return &GenericScaler{scalesGetter, gr}, nil
-	}
+func ScalerFor(scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) Scaler {
```
Function is no longer needed.

deads2k commented on test/e2e/framework/util.go:
```diff
@@ -2683,16 +2683,12 @@ func RemoveAvoidPodsOffNode(c clientset.Interface, nodeName string) {
-//TODO(p0lyn0mial): remove internalClientset and kind
-func getScalerForKind(internalClientset internalclientset.Interface, kind schema.GroupKind, scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) (kubectl.Scaler, error) {
-	return kubectl.ScalerFor(kind, internalClientset, scalesGetter, gr)
+func getScalerForKind(scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) kubectl.Scaler {
```
Function is no longer needed

deads2k commented:
@soltysh What's up with this error: 

`error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could not find the requested resource`

p0lyn0mial commented:
> What's up with this error:
>
> error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could > not find the requested resource

Does a Job resource have a Scale subresource ?

deads2k commented:
> What's up with this error:
> 
> error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could > not find the requested resource
> Does a Job resource have a Scale subresource ?

No it doesn't.  @soltysh naughty naughty naughty.  You'll need to add that.

deads2k commented:
@p0lyn0mial it's going to take me a little while to get the jobs endpoint sorted out.  Can you replace the others?

p0lyn0mial commented:
commits squashed, vet fixed.
we have to wait for @soltysh  pull to be merged before we can squash that one in.

p0lyn0mial committed:
removes the remainder from ScalerFor method

all remaining scalers were replaced by GenericScaler exept JobScaler.
It is not clear whether JobScaler could use generic scaler or not.
For more details see the pull request.

p0lyn0mial commented:
/restest all

deads2k commented:
gofmt

p0lyn0mial commented:
> gofmt

believe or not but I ran `hack/update-all` :)

deads2k commented:
/retest

deads2k commented:
/lgtm

Random-Liu commented:
Also https://github.com/kubernetes/kubernetes/issues/59058.

Agree with reverting this.
